### PR TITLE
feat(core): Defer task on launcher handshake (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
## Summary

This PR introduces the `runner:taskdeferred` message type to allow the launcher to ask the broker to hold onto a task until the launcher can spin up a runner and this runner can connect, register itself, and send its first task offer, allowing the launcher to spawn a runner only when needed.

```mermaid
sequenceDiagram
    participant TM as Task Manager
    participant TB as Task Broker
    participant L as Launcher
    participant TR as Task Runner

    %% Initial launcher connection
    L->>+TB: WS connect
    TB-->>L: broker:info-request
    L->>TB: runner:info
    TB-->>L: broker:runner-registered

    %% Task request flow
    L->>TB: runner:task-offer
    TM->>+TB: requester:task-request
    TB-->>L: broker:task-offer-accept
    L->>TB: runner:task-deferred

    %% Launcher executes and transitions to runner
    Note over L: exec.Command()
    
    %% New runner connection
    TR->>+TB: WS connect
    TB-->>TR: broker:info-request
    TR->>TB: runner:info
    TB-->>TR: broker:runner-registered
    TR->>TB: runner:task-offer
    TB-->>L: broker:task-offer-accept
    TB-->>TM: broker:task-ready

    %% Task settings flow
    TM->>+TB: requester:task-settings
    TB-->>TR: broker:task-settings
    TR->>TB: runner:task-done
    TB-->>TM: broker:task-done
    TR->>TR: exit if idle
```

To be reviewed together with: https://github.com/n8n-io/task-runner-launcher/pull/6

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/PAY-2244/implement-handshake-with-main

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/runners/task-broker.service.ts [hunk 1]
```diff
@@ -223,6 +233,14 @@ export class TaskBroker {
 		}
 	}
 
+	handleRunnerDeferred(taskId: Task['id']) {
+		const acceptReject = this.runnerAcceptRejects.get(taskId);
+		if (acceptReject) {
+			acceptReject.reject(new TaskDeferredError());
+			this.runnerAcceptRejects.delete(taskId);
+		}
```
Comment (tomi) [inline packages/cli/src/runners/task-broker.service.ts:241]: I don't think we actually need to reject here. We could push the task directly back to pending tasks

Comment (ivov) [reply packages/cli/src/runners/task-broker.service.ts:241]: Gave this a try and it kind of works, but I'm not satisfied - we need to add more properties to messages on both sides, add an exception to a timeout, and also something about this is [preventing main from exiting cleanly](https://share.cleanshot.com/26MRtMFV).

See these branches:

- `main/handshake-tomi-feedback`
- `launcher/handshake-tomi-feedback`

Let's discuss tomorrow.

Comment (ivov) [reply packages/cli/src/runners/task-broker.service.ts:241]: > something about this is [preventing main from exiting cleanly](https://share.cleanshot.com/26MRtMFV)

This is actually broken on `master` → https://github.com/n8n-io/n8n/pull/11801

Comment (tomi) [reply packages/cli/src/runners/task-broker.service.ts:241]: Oh right you are absolutely right. I didn't realize that `acceptOffer` does a bunch of stuff that must be handled as well. The way you implemented it is actually really nice of handling this 👌

**Review by tomi — APPROVED** [review]
Really nice work. The diagram made this a lot easier to follow ❤️  🚀

**Pull request merged**